### DISCUSSION TITLE
Cannot tell if the file scan is getting hung up or just taking a long time

### DISCUSSION THREAD

**Discussion by PeterUSC** [qarmin/czkawka#1234] [General]
So I am trying to get rid of duplicates in huge folders (1TB-2TB of photos). To make it worse it is on a hard drive rather an SSD. 

When starting a search the GUI it says "Scanning size of 1000 file" then quickly "Scanning size of 2534 file" and etc. At some point it gets to some number in the millions and seems to get hung up. I cannot tell if it is still working and is just going to take a while or it has stopped. Waited for 30 mins and it was still seemingly hung up and a attempt to stop search appears to hang up to. 

is there a way to tell if it is still searching? Maybe see the name of the file it is "scanning the size" of?

**Comment by qarmin** [comment]
30 minutes is too much.
I suspect that app could crash.
Can you run app via CLI with debug flags? 
```
RUST_LOG=debug ./czkawka
```

Progress is send with format `files to check/ files checked` + `current mode` + `current step / all steps`. Code is multithreaded, so adding there info about current tested files could be problematic.

**Comment by PeterUSC** [comment]
This what you are looking for? got hung up on the collecting files. 
```
RUST_LOG=debug ./czkawka_cli dup -T 8  -m 24 -d /DUPLICATES
11:08:47.850 [INFO] czkawka_core::common: App version: 6.1.0, release mode, rust 1.76.0 (2024-02-04), os Mac OS 14.3.0 [aarch64 64-bit], 8 cpu/threads
11:08:47.851 [DEBUG] czkawka_core::common: Number of threads set to 8
11:08:47.851 [INFO] czkawka_core::duplicate: find_duplicates
11:08:47.851 [DEBUG] czkawka_core::duplicate: check_files_size
11:08:47.852 [DEBUG] czkawka_core::common_dir_traversal: run(collecting files/dirs)
Collecting files: 1042064 ▹▹▹▹▸                                                                                                                                                    
11:09:54.346 [DEBUG] czkawka_core::common: send_info_and_wait_for_ending_all_threads
11:09:54.354 [DEBUG] czkawka_core::common: send_info_and_wait_for_ending_all_threads: Done in 7.80ms
11:09:54.355 [DEBUG] czkawka_core::common_dir_traversal: Collected 1054117 files
11:09:54.355 [DEBUG] czkawka_core::common_dir_traversal: run(collecting files/dirs): Done in 66.51s
Collecting files: 1042064 ▹▸▹▹▹    
```

**Reply by qarmin** [reply]
Looks that second message `Collecting files: 1042064 ▹▸▹▹▹    ` is not proper(probably CLI somehow cached it) and app hands on finding duplicates or loading data from cache - but still should almost instantly show different message

**Reply by PeterUSC** [reply]
are there issues with directories of very large sizes? 

I tried running it on three separate directories. Directory_1 size totals to 2.25TB,  Directory_2 size totals to 2.21TB, and Directory_3 totals to 1.02TB. 

Duplication searches on Directory_1 and Directory_2 lead to the hang up problem. However, the same search of Directory_3 works. The size of the directories is the biggest difference between them

**Comment by PeterUSC** [comment]
Sorry accidentally closed the issue
